index.js

```diff
@@ -115,15 +116,24 @@ exports = module.exports = function morgan(format, options) {
         return
       }
 
-      var line = fmt(exports, req, res)
-
+      var formatted = fmt(exports, req, res);
+      var line;
+      var tokenPairs;
+      
+      if (typeof formatted === 'object') {
+        line = formatted.line;
+        tokenPairs = formatted.tokenPairs
+      } else {
+        line = formatted;
+      }
+      
       if (null == line) {
         debug('skip line')
         return
       }
 
-      debug('log request')
-      stream.write(line + '\n')
+      debug('log request');
+      includeTokens ? stream.write(line + '\n', tokenPairs) : stream.write(line + '\n');
```

Comment:
This is not valid, unfortunately. The second argument to `stream.write` already is defined: it's encoding, you cannot use it for something else.